pkg/route/validation/validation.go

```diff
@@ -342,6 +382,56 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
+	notAllowedHTTPHeaderSet = sets.NewString(notAllowedHTTPHeaders...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	unique := make(map[string]bool, len(headerValues))
+	us := make([]string, len(unique))
+	for key, value := range headerValues {
+		//for _, elem := range s {
+		if len(value.Name) != 0 {
+			if !unique[value.Name] {
+				us = append(us, value.Name)
+				unique[value.Name] = true
+			} else {
+				return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("header name has been repeated"))
+			}
+		}
+		//}
+
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 && len(value.Action.Set.Value) > 16384 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, fmt.Sprintf("value exceeds the maximum length which is 16384"))
+		}
+		if len(value.Name) != 0 && len(value.Name) > 1024 {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("name exceeds the maximum length which is 1024"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Action.Set.Value, fmt.Sprintf("value must be present"))
+		}
+		if len(value.Name) == 0 {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("name must be present"))
+		}
+		if path.String() == specResponseString && value.Action.Set != nil && len(value.Action.Set.Value) != 0 {
+			if !permittedResponseHeaderValueRE.MatchString(value.Action.Set.Value) {
+				return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, "Either header value provided is not in correct format or the converter specified is not allowed. The dynamic header value  may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are res.hdr, ssl_c_der. Converters allowed are lower, base64.")
+			}
+		}
+		if path.String() == specRequestString && value.Action.Set != nil && len(value.Action.Set.Value) != 0 {
+			if !permittedRequestHeaderValueRE.MatchString(value.Action.Set.Value) {
+				return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, "Either header value provided is not in correct format or the converter specified is not allowed. The dynamic header value  may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2 Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64.")
+			}
+		}
+		if len(value.Name) != 0 && value.Action.Set != nil && notAllowedHTTPHeaderSet.Has(strings.ToLower(value.Name)) {
+			return field.NotSupported(path.Index(key).Child("name"), value.Name, []string{fmt.Sprintf("All headers except %v", strings.Join(notAllowedHTTPHeaders, ","))})
```

Comment:
What does this print out as an error?  I question the "All headers except ..."

Reply:
errorMessage.Field spec.httpHeaders.actions.response[0].name: Unsupported value: "Strict-Transport-Security": supported values: "All headers except strict-transport-security,proxy,cookie,set-cookie"